x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_enrollment_flyout/agent_policy_selection.tsx

```diff
@@ -204,28 +211,89 @@ export const EnrollmentStepAgentPolicy: React.FC<Props> = (props) => {
           {isAuthenticationSettingsOpen && (
             <>
               <EuiSpacer size="m" />
-              <EuiSelect
-                fullWidth
-                options={enrollmentAPIKeys.map((key) => ({
-                  value: key.id,
-                  text: key.name,
-                }))}
-                value={selectedState.enrollmentAPIKeyId || undefined}
-                prepend={
-                  <EuiText>
+              {enrollmentAPIKeys.length && selectedState.enrollmentAPIKeyId ? (
+                <EuiSelect
+                  fullWidth
+                  options={enrollmentAPIKeys.map((key) => ({
+                    value: key.id,
+                    text: key.name,
+                  }))}
+                  value={selectedState.enrollmentAPIKeyId || undefined}
+                  prepend={
+                    <EuiText>
+                      <FormattedMessage
+                        id="xpack.fleet.enrollmentStepAgentPolicy.enrollmentTokenSelectLabel"
+                        defaultMessage="Enrollment token"
+                      />
+                    </EuiText>
+                  }
+                  onChange={(e) => {
+                    setSelectedState({
+                      ...selectedState,
+                      enrollmentAPIKeyId: e.target.value,
+                    });
+                  }}
+                />
+              ) : (
+                <EuiCallOut
+                  color="warning"
+                  title={i18n.translate(
+                    'xpack.fleet.agentEnrollment.noEnrollmentTokensForSelectedPolicyCallout',
+                    {
+                      defaultMessage:
+                        'There are no enrollment tokens for the selected agent policy',
+                    }
+                  )}
+                >
+                  <div className="eui-textBreakWord">
+                    <FormattedMessage
+                      id="xpack.fleet.agentEnrollment.noEnrollmentTokensForSelectedPolicyCalloutDescription"
+                      defaultMessage="You must create and enrollment token in order to enroll agents with this policy"
+                    />
+                  </div>
+                  <EuiSpacer size="m" />
+                  <EuiButton
+                    iconType="plusInCircle"
+                    isLoading={isLoadingEnrollmentKey}
+                    fill
+                    onClick={() => {
+                      setIsLoadingEnrollmentKey(true);
+                      if (selectedState.agentPolicyId) {
+                        sendCreateEnrollmentAPIKey({ policy_id: selectedState.agentPolicyId })
```

Comment:
should we add a `Default` name as we have when we create the token by default for the user?

Reply:
ping @hbharding fyi but SGTM

Reply:
Actually, we'll run into an issue here if we

1. Click button to create token (creates `Default (aaaa-bbbb-cccc-dddd)`)
2. Revoke token from (1)
3. Repeat (1)
4. See error that 

<img width="1427" alt="Screen Shot 2021-05-25 at 6 14 40 PM" src="https://user-images.githubusercontent.com/57655/119575702-72d82500-bd85-11eb-9144-c4266241a075.png">

Reply:
Ahh, shoot. We use `Default` as the enrollment token name behind the scenes when users create an agent policy (this triggers the enrollment token to be created). I thought if users revoked that one, we would be able to create another called `Default`, but I was wrong to assume that the ID appended to the name was random (it's not, seems related to the agent policy). If `Default` already existed as a name, would it be possible to increment the name to be `Default-2`, etc.?

Reply:
> but I was wrong to assume that the ID appended to the name was random (it's not, seems related to the agent policy)

The ID is random. The issue is that the uniqueness check specifically ignores the ` (some-uuid-goes-here)` portion and only looks at the given name (e.g. `Default`).

> If `Default` already existed as a name, would it be possible to increment the name to be `Default-2`, etc.?

Yes, I don't think that's too difficult. I'll give it a shot and report back.

Reply:
@hbharding  I quickly ran into an issue when returning to this.

If we create `Default-2` and the user later invalidates _that_ we're back to facing the same issue. We could try `Default-3`, but I think you see where this is going.

Rather than dealing with those UX & eng issues, I think we should either:

  1. Not specify a name (current PR status)
  2. Include an input so the user can supply one

I guess another alternative is 

 3. Try `Default` and if that fails display (2)

Reply:
> Not specify a name (current PR status)

What happens in this case? No name is given to the enrollment token, so it is possible to create another enrollment token with "no name"?

Reply:
> > Not specify a name (current PR status)
>
> What happens in this case? No name is given to the enrollment token, so it is possible to create another enrollment token with "no name"?

Yes. You can see the interaction in the second video in the description. There is a unique name constraint but name is an optional property so omitting the name avoids the constraint.

Digging into the details a bit more, we actually do always name the key. The API & UI will have a value for the key. The code to determine the name looks like:

```
const keyName = userProvidedKeyName ? `${userProvidedKeyName} (${uuid})` : uuid;
```

Looking at the screenshot below
 * the first six rows are the UUID-only form (created by the service)
 * the last two were given a name so they are `Given name (some-uuid-goes-here)`
<img width="494" alt="Screen Shot 2021-06-03 at 11 48 29 AM" src="https://user-images.githubusercontent.com/57655/120673975-a5f37600-c461-11eb-8256-549d5a51258b.png">

Reply:
Let's go ahead and get this in without using any name. I expect this to be an edge case anyway.

Reply:
Sorry for the delay. Thanks @jfsii. I'm leaning towards (2) - include an input - as it seems simple and failproof. Using a "nameless" token (1) seems a little odd and might cause confusion on the token table ("why are some of these named but others aren't?"). Also, what would happen if users delete a nameless token and then try to create a new nameless token... i suspect the same naming issue? :)

Here's a quick mock to guide you. It felt strange to have an input inside of the callout, so I'm suggesting we move the "create enrollment token" button outside of the callout to sit with the text input.

![image](https://user-images.githubusercontent.com/847805/120846675-79faf200-c540-11eb-90cf-35e3f702a3c3.png)

Reply:
Whoops - I missed the comments that came in before I posted. I'm fine with using (1). I checked with @jfsiii and learned having multiple "nameless" tokens does not cause an issue.